deps/jerry/v8jerry/v8jerry_value.cpp

```diff
@@ -81,13 +81,23 @@ JerryValue* JerryValue::GetInternalProperty(JerryValue* key) {
 
 JerryValue* JerryValue::GetOwnPropertyDescriptor(const JerryValue& jkey) const {
     jerry_value_t key = jkey.value();
-    jerry_value_t descriptor = JerryIsolate::GetCurrent()->HelperGetOwnPropDesc().Call(m_value, &key, 1);
-
+    jerry_property_descriptor_t prop_desc;
+    jerry_get_own_property_descriptor (m_value, key, &prop_desc);
+    jerry_value_t descriptor = jerry_from_property_descriptor (&prop_desc);
     return new JerryValue(descriptor);
 }
 
 JerryValue* JerryValue::GetOwnPropertyNames() const {
-    return new JerryValue(JerryIsolate::GetCurrent()->HelperGetOwnPropNames().Call(m_value, NULL, 0));
+    jerry_property_filter_t flags = (jerry_property_filter_t) (JERRY_PROPERTY_FILTER_EXLCUDE_SYMBOLS
+                                                               | JERRY_PROPERTY_FILTER_EXLCUDE_NON_ENUMERABLE);
+    jerry_value_t property_names_val = jerry_object_get_property_names (m_value, flags);
+
+    if(jerry_value_is_array (property_names_val))
```

Comment:
If this condition does not evaluates to true I think an empty local value should be returned in `api.cc`. I think this helper method can be removed.